### ISSUE TITLE
UOAIS-net Model Training Related Query: (Freezing layers within model architecture & changing model training hyperparameters)

### ISSUE THREAD

**Issue by llvone** [ngzhili/uoais-v2#1]
Hello @ngzhili @HZWang96 @Trav1slaflame, thank you very much for sharing how the SynTable pipeline can be applied to generate synthetic dataset for application within training & evaluation of the UOAIS-net model! 

I am working towards applying synthetic amodal instance segmentation dataset as input within model training of pre-trained UOAIS-net model weight. **I was hoping to seek your possible feedback regarding how to change certain hyperparameters/architecture of Resnet50 model(which compose the RGB backbone & Depth backbone) during model training.**

### **Freezing residual blocks of Resnet-50 backbone during model training of pre-trained UOAIS-net model weight:**
Specifically, I wanted to seek possible feedback for steps to freeze the residual blocks of the RGB and depth backbones (Resnet-50). 

I realize that the backbone residual blocks can be frozen by **defining MODEL.BACKBONE.FREEZE_AT = (VALUE)** within the config .yaml file(R50_rgbdconcat_mlc_occatmask_hom_concat.yaml). However, it seems that the model's specified residual blocks did not freeze during model training despite applying the freeze parameter within the .yaml file. 

I did try to freeze the residual blocks of the pre-trained UOAIS-net model using other steps; however, there was no indication of the specified residual blocks actually being frozen during the UOAIS model training. 

**Other steps include:** 
1. Directly specifying the command **(cfg.MODEL.BACKBONE.FREEZE_AT = (VALUE))** within the setup function (lines 244-258) of the train_net.py script within the uoais repository. 

2. Defining **"freeze_at = (value)" as input cfg parameter within the build_resnet_backbone function** (line 418) and build_depth_resnet_backbone(line 420) within the function(build_resnet_rgbd_latefusion_fpn_backbone) of script(rgbdfpn.py). 

I also tested model training with an invalid instance of freeze residual block parameter to confirm if the model training would successfully run or fail to execute. 

For instance, I applied **cfg.MODEL.BACKBONE.FREEZE_AT = 50** (to freeze 50 residual blocks within the pre-trained UOAIS model backbone) within config .yaml file. As the rgb backbone and the depth backbone do not consist of 50 residual blocks, it would seem the model training should not theoretically initialize. 

_However, despite the number of freeze residual blocks defined as 50, the UOAIS-net model training did initialize & execute for an invalid instance.  
Therefore, I wanted to request possible feedback on how to freeze the residual blocks of the model._

### **Registering the Resnet backbone using the model training parameters config .yaml file (R50_rgbdconcat_mlc_occatmask_hom_concat.yaml):**

**- I am wondering if the model training specifications outlined within the model training config .yaml file(R50_rgbdconcat_mlc_occatmask_hom_concat.yaml) are applicable to a UOAIS-net model trained from scratch only or would these training specifications be applicable for a pre-trained UOAIS-net model weight as well?**

I am asking as within the R50_rgbdconcat_mlc_occatmask_hom_concat.yaml file, the MODEL.BACKBONE.NAME parameter calls the function(build_resnet_rgbd_latefusion_fpn_backbone). As this function(build_resnet_rgbd_latefusion_fpn_backbone)defined within the script(rgbdfpn.py) calls the function [(build_resnet_backbone) - in line 418)], which creates a ResNet instance from config. 

**Therefore, if applying a pre-trained UOAIS-net model weight to initialize model training, should this function/model name (build_resnet_rgbd_latefusion_fpn_backbone) be called in the .yaml config file as this function creates a new ResNet instance from config?** 

This function(build_resnet_backbone) is defined within the external detectron2's source code (Specifically: from detectron2.modeling.backbone.resnet) and is called to include as part of the RGB backbone. 

I would greatly appreciate your any possible feedback and thank you very much for your time!

**Comment by HZWang96** [comment]
Hi @llvone, I'm sorry for the late reply. Unfortunately, we are not the best people to answer this question. We did not make any modifications to the network architectures of the models that were cited in our paper. We trained and evaluated all models as specified by the authors of UOAIS-Net in their [GitHub repo](https://github.com/gist-ailab/uoais). In this repo, we only added the code to evaluate the occlusion order accuracy of the UOAIS-Net model.

I see that you have already posted the same issue in the UOAIS-Net repo as well. I hope the issue gets resolved soon!
